grails-app/services/com/unifina/service/CommunityJoinRequestService.groovy

```diff
@@ -0,0 +1,132 @@
+package com.unifina.service
+
+import com.streamr.client.protocol.message_layer.StreamMessage
+import com.unifina.api.ApiException
+import com.unifina.api.CommunityJoinRequestCommand
+import com.unifina.api.NotFoundException
+import com.unifina.api.UpdateCommunityJoinRequestCommand
+import com.unifina.domain.community.CommunityJoinRequest
+import com.unifina.domain.community.CommunitySecret
+import com.unifina.domain.data.Stream
+import com.unifina.domain.security.IntegrationKey
+import com.unifina.domain.security.SecUser
+import com.unifina.signalpath.utils.MessageChainUtil
+import com.unifina.utils.MapTraversal
+import grails.util.Holders
+import org.web3j.crypto.Credentials
+
+class CommunityJoinRequestService {
+	StreamService streamService
+	EthereumService ethereumService
+
+	private void onApproveJoinRequest(CommunityJoinRequest c) {
+		String joinPartStreamID = ethereumService.fetchJoinPartStreamID(c.communityAddress)
+		Map<String, Object> msg = new HashMap<>()
+		msg.put("type", "join")
+		msg.put("addresses", Arrays.asList(c.memberAddress))
+		sendMessage(joinPartStreamID, msg)
```

Comment:
Permission granting is missing from here. I'll file a separate JIRA ticket about that in order get this PR merged, so that the frontend guys can begin work on the CP frontend.